### ISSUE TITLE
how to compute "Best SDL"

### ISSUE THREAD

**Issue by ruili-pml** [VICO-UoE/URL#15]
Hi,

I was wondering do you still have the code for computing the "Best SDL" in Table 1 in the URL paper? Or alternatively, could you help with the following  problem:

When I was trying to assess the performance of single domain pretrained NN the labels returned by test loader seems strange. If my understanding is correct, then by

```python
test_loader = MetaDatasetEpisodeReader('test', test_set = ['vgg_flower'], test_type=args['test.type'])
config = tf.compat.v1.ConfigProto()
config.gpu_options.allow_growth = False

with tf.compat.v1.Session(config=config) as session:
    
    test_sample = test_loader.get_test_task(session)
```

```test_sample['target_gt']``` should return the original label for the target set. To assess pretrained SDL NN, I just need to use it to make the prediction and calculate the accuracies. But ```test_sample['target_gt']``` doesn't seem to be the original label for the target set (in vgg flower there're only 71 classes but when I print out ```test_sample['target_gt']``` the smallest number is larger than 80)? I was wondering if I am i missing something here?

Many thanks in advance.

**Comment by WeiHongLee** [comment]
Hi,

Thanks for the questions. For computing the "Best SDL" as in Table 1 in the URL paper, as stated in the paper, we use each single-domain network as the feature extractor and test it for few-shot classification in each dataset and pick the best performing model. This involves evaluating 8 single-domain networks on 13 datasets. The results of each network are similar to the ones in Table 2 in the [supplementary material](https://openaccess.thecvf.com/content/ICCV2021/supplemental/Li_Universal_Representation_Learning_ICCV_2021_supplemental.pdf). If you would like to compute them by yourself, you can simply use `test_extractor.py` and evaluate all networks on each testing set and pick the best performance for each testing set.

`test_sample['target_gt']` returns the ground truth labels for the original datasets, e.g. vgg_flower has 102 classes in total as listed [here](https://github.com/google-research/meta-dataset#dataset-summary). If you would like to evaluate the meta-test performance, the labels should be `test_sample['target_labels']`. I would suggest to adapt `test_extractor.py` for evaluation.

Best!

**Comment by WeiHongLee** [comment]
Hi,

Results of "Best SDL" is computed by evaluating all SDL on every testing set using SDL pre-trained model (frozen) + Nearest Centroid Classifier (NCC). This assesses the pre-trained feature extractor for seen and unseen domains. We do not use the pre-trained linear classifiers for evaluation.

Best!

**Comment by ruili-pml** [comment]
Got it, thank you very much for the quick reply :)
